### DISCUSSION TITLE
Persistent layout with Vue 3.2 setup script

### DISCUSSION THREAD

**Discussion by ibrunotome** [inertiajs/inertia#873] [Help (Vue)] [answered]
### Versions:

- `@inertiajs/inertia` version: 0.10.1
- `@inertiajs/inertia-vue3` version: 0.5.2

### Describe the problem:

In the docs the persistent layout in vue 3 is easy with 

```vue
<script lang="ts">
  import Layout from './Layout'

  export default {

    layout: Layout,
    
  }
</script>
```

How to set persistent layout in [new vue 3.2 setup script](https://blog.vuejs.org/posts/vue-3.2.html) that doesn't need the export default section?

```vue
<script lang="ts" setup>
  import Layout from './Layout' // after the import, the Layout is already available to be used in <template>
</script>
```

**Comment by ajnsn** [comment, marked as answer]
@ibrunotome [Vue Docs](https://v3.vuejs.org/api/sfc-script-setup.html#usage-alongside-normal-script): `<script setup>` can be used alongside normal `<script>`, this should work?

```vue
<script>
export default {
  layout: Layout,
}
</script>

<script setup>
...
</script>
```

**Reply by ibrunotome** [reply]
@ajnsn yes, I started to use this way yesterday 👍
